all-the-icons-dired.el

```diff
@@ -57,92 +56,43 @@
-                                      `(:v-adjust ,all-the-icons-dired-v-adjust)
-                                      (when all-the-icons-dired-monochrome
-                                        `(:face ,(face-at-point))))))))
-            (if (member file '("." ".."))
-                (all-the-icons-dired--add-overlay (dired-move-to-filename) "  \t")
-              (all-the-icons-dired--add-overlay (dired-move-to-filename) (concat icon "\t"))))))
-      (forward-line 1))))
-
-(defun all-the-icons-dired--refresh-advice (fn &rest args)
-  "Advice function for FN with ARGS."
-  (prog1 (apply fn args)
-    (when all-the-icons-dired-mode
-      (all-the-icons-dired--refresh))))
-
-(defvar all-the-icons-dired-advice-alist
-  '((dired-aux     dired-do-redisplay           all-the-icons-dired--refresh-advice)
-    (dired-aux     dired-create-directory       all-the-icons-dired--refresh-advice)
-    (dired-aux     dired-do-create-files        all-the-icons-dired--refresh-advice)
-    (dired-aux     dired-do-kill-lines          all-the-icons-dired--refresh-advice)
-    (dired-aux     dired-do-rename              all-the-icons-dired--refresh-advice)
-    (dired-aux     dired-insert-subdir          all-the-icons-dired--refresh-advice)
-    (dired-aux     dired-kill-subdir            all-the-icons-dired--refresh-advice)
-    (dired         wdired-abort-changes         all-the-icons-dired--refresh-advice)
-    (dired         dired-internal-do-deletions  all-the-icons-dired--refresh-advice)
-    (dired-narrow  dired-narrow--internal       all-the-icons-dired--refresh-advice)
-    (dired-subtree dired-subtree-insert         all-the-icons-dired--refresh-advice)
-    (dired-subtree dired-subtree-remove         all-the-icons-dired--refresh-advice)
-    (dired         dired-readin                 all-the-icons-dired--refresh-advice)
-    (dired         dired-revert                 all-the-icons-dired--refresh-advice)
-    (find-dired    find-dired-sentinel          all-the-icons-dired--refresh-advice))
-  "A list of file, adviced function, and advice function.")
+(defun all-the-icons-dired--put-icon (pos) "Propertize POS with icon."
+       (let* ((file (dired-get-filename 'relative 'noerror))
+              (icon (if (file-directory-p file)
+                        (all-the-icons-icon-for-dir file
+                                                    :face 'all-the-icons-dired-dir-face
+                                                    :v-adjust all-the-icons-dired-v-adjust)
+                      (apply 'all-the-icons-icon-for-file file
+                             (append
+                              `(:v-adjust ,all-the-icons-dired-v-adjust)
+                              (when all-the-icons-dired-monochrome
+                                `(:face ,(face-at-point))))))))
+         (put-text-property (1- pos) pos 'display
+                            (if (member file '("." "..")) "    " (concat " " icon " ")))))
```

Comment:
A stylistic point: I find it easier to read an `if` that spans multiple lines when it includes balanced expressions inside of it.